### PR TITLE
Add serializer for registration view

### PR DESCRIPTION
**Pull request by k9845**
Addresses 
- https://github.com/IFRCGo/go-api/issues/1839#event-9719319811 

## Changes
- Add serializer for registration view

## Checklist 
Things that should succeed before merging.

- [x] Updated/ran unit tests
- [ ] Updated CHANGELOG.md

## Release

If there is a version update, make sure to tag the repository with the latest version.

### CODE REVIEW

File: registrations/serializers.py [hunk 1]
```diff
@@ -1,9 +1,118 @@
...
+                country,
+                organization_type,
+                organization,
+                city,
+                department,
+                position,
+                phone_number
+            )
+        except Exception:
+            User.objects.filter(username=username).delete()
```
Comment (thenav56) [inline registrations/serializers.py:98]: Not required, this should be handled by transaction

File: registrations/serializers.py [hunk 2]
```diff
@@ -1,9 +1,118 @@
...
+            return serializers.ValidationError('Could not create user profile.')
+
+        pending = Pending.objects.create(
+            user=user,
+            justification=justification,
+            token=get_random_string(length=32)
+        )
+        if not is_staff:
+            pending.admin_token_1 = get_random_string(length=32)
+        pending.save()
```
Comment (thenav56) [inline registrations/serializers.py:108]:
```suggestion
        pending = Pending(
            user=user,
            justification=justification,
            token=get_random_string(length=32)
        )
        if not is_staff:
            pending.admin_token_1 = get_random_string(length=32)
        pending.save()
```

File: registrations/test_views.py [hunk 3]
```diff
@@ -99,13 +103,15 @@ def test_official_email(self):
...
-            'firstname': 'Peter',
-            'lastname': 'Falk',
+            'first_name': 'Peter',
+            'last_name': 'Falk',
+            'justification': 'aaaa',
+            'city': 'kathmandu'
         }
         headers = {'CONTENT_TYPE': 'application/json'}
-        resp = self.client.post('/register', body, format='json', headers=headers)
+        resp = self.client.post('/register', data=body, format='json')
```
Comment (thenav56) [inline registrations/test_views.py:114]: Maybe this works
```suggestion
        resp = self.client.post('/register', json=body)
```

File: registrations/test_views.py [hunk 4]
```diff
@@ -124,3 +130,36 @@ def test_official_email(self):
         # 5b. Confirming that a user with an official email is activated
         boarded_user = User.objects.get(username=newusr)
         self.assertTrue(boarded_user.is_active)
+
+    @mock.patch('registrations.serializers.send_notification_create')
+    def test_user_registration(self, send_notification_create):
+        country = Country.objects.get(name='country')
```
Comment (thenav56) [inline registrations/test_views.py:136]: Please confirm this.

File: registrations/test_views.py [hunk 5]
```diff
@@ -124,3 +130,36 @@ def test_official_email(self):
...
+            'country': country.pk,
+            'organization_type': 'OTHR',
+            'organization': 'Zoo',
+            'first_name': 'Peter',
+            'last_name': 'Falk',
+            'justification': 'aaaa',
+            'city': 'kathmandu'
+        }
+        resp = self.client.post('/register', data, format='json')
+        self.assertEqual(resp.status_code, 400)
```
Comment (thenav56) [inline registrations/test_views.py:153]: Let's assert for user count as well and error response.

File: registrations/utils.py [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+    user.profile.org = organization
+    user.profile.city = city
+    user.profile.department = department
+    user.profile.position = position
+    user.profile.phone_number = phone_number
+    user.save()
+    return user
+
+
+def is_valid_domain(email):
```
Comment (thenav56) [inline registrations/utils.py:50]: Add a note to update the function definition.

**Review by thenav56 — CHANGES_REQUESTED** [review]

File: registrations/serializers.py [hunk 7]
```diff
@@ -1,9 +1,118 @@
...
         fields = '__all__'
+
+
+class RegistrationSerializer(serializers.Serializer):
+    # required fields
+    email = serializers.CharField()
+    password = serializers.CharField()
+    first_name = serializers.CharField()
+    last_name = serializers.CharField()
+    country = serializers.IntegerField()
```
Comment (thenav56) [inline registrations/serializers.py:28]: Let's define max_length here as well.

**Review by thenav56 — CHANGES_REQUESTED** [review]

**Comment by k9845** [comment]
Closing in favour of `https://github.com/IFRCGo/go-api/pull/1846`

**Comment by k9845** [comment]
Closing in favour of `https://github.com/IFRCGo/go-api/pull/1846`

**Pull request closed without merging**